src/torrent_handle.cpp

```diff
@@ -361,6 +361,12 @@ namespace libtorrent
 		TORRENT_ASYNC_CALL(queue_down);
 	}
 
+	void torrent_handle::queue_position_set(int p) const
+	{
+		TORRENT_ASSERT_PRECOND(p >= 0);
```

Comment:
when built in release mode, you should have a second check too. ``if (p < 0) return;``